oak/common/app_config.bzl

```
@@ -0,0 +1,40 @@
+#
+# Copyright 2020 The Project Oak Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Rule for serializing an Oak application configuration."""
+
+def serialized_config(name, textproto, modules):
+    """Serializes an Oak application configuration in a binary file.
+
+    Args:
+        name: Name of the build rule.
+        textproto: Textproto file with application configuration.
+        modules: A dictionary with module names as keys and module paths as values.
+    """
+    srcs = [textproto] + modules.values()
+    output = name + ".bin"
+    module_list = ",".join([n + ":$(location " + path + ")" for (n, path) in modules.items()])
+    cmd = "$(location //oak/common:app_config_serializer)" + \
+          " --textproto=$(location {})".format(textproto) + \
```

Would this work?

```suggestion
          " --textproto=$(location {textproto})"
```

From https://www.programiz.com/python-programming/string-interpolation

---

IIUC it's only possible in `f-strings` in python3.6, and it's probably not implemented in Starlark.
Doesn't seem to work - `f' --modules={module_list}'` gives:
```
ERROR: /opt/my-project/oak/common/app_config.bzl:31:11: name 'f' is not defined
```
and `" --modules={module_list}"` gives:
```
label '//examples/hello_world/config:{textproto}' in $(location) expression is not a declared prerequisite of this rule.
```